docs/development_suite/set-up-infrastructure/enhanced-project-workflow.md

```diff
@@ -0,0 +1,136 @@
+- The new `environments` directory is automatically created by the Console. Inside it, the generated configurations for each environment are stored in separate directories. 
+:::warning
+It is important to note that these are auto-generated directories, so all their contents are regenerated each time a push operation is performed. If you add custom files to the environments directory or its child directories, they will be deleted. 
+:::
+
+- If you are **deploying without Kustomize**, you will find a `configuration` directory containing one directory for each environment, and possibly some custom files.
+You need to create the overlays directory, where you have to move environment directories previously stored inside `configuration`. No actions are required for the custom files in the `configuration` directory. 
+Note: In case a custom file with the same name is present inside both the `configuration` directory and an environment directory, the global custom file will prevail.
+- If you are **using Kustomize**, the `configuration` directory contains a `kustomization.yaml` manifest and possibly your custom files, which are already in the right spot. You can delete every other file inside the directory.
+
+    Also, make sure that: 
+    - Inside each `environments/<environment>` directory, the `kustomization.yaml` manifest imports the `kustomization.yaml` manifest from the `configuration` directory. <!-- This should be automatically handled by the Console. -->
+    - Inside each `overlays/<environment>` directory, the `kustomization.yaml` manifest imports the corresponding manifest from the `environments/<environment>` directory.
+
+## Migrating existing projects
+
+In order to adopt the enhanced workflow on existing projects, a migration process is required.
+
+<!-- here we are talking about moving files to a database. Is it ok? -->
+All branches and tags need to be migrated to a database in the form of revisions and versions. For each revision and version, the Console configuration files must be moved to specific documents on the database, each one associated with its `configType` property. These files are: 
+
+- `api-console-config.json` (merged with all ConfigMaps content) - `configType: base`
+- `fast-data-config.json` - `configType: fast-data`
+- `rbacManagerConfig.json` - `configType: rbacManagerConfig`
+- `backofficeConfiguration.json` - `configType: backofficeConfigurations`
+- `mia-craft-config.json` - `configType: ` <!-- TBD -->
+- All config extension files - `configType: extensions`
+
+The project data model has to be updated by ensuring that the `defaultBranch` property corresponds to the repository’s default branch, or to a fallback value.
+
+Moreover, the Git repository structure has to be re-shaped to complete the migration process.
+
+The final repository structure will be the following: 
```

Comment:
This info could be placed also to "## Manual adjustments on newly created projects" paragraph to explain better how the git repo is strctured after the creation of a 'new' proejct